test/gtest/mhp/zip-base.cpp

```diff
@@ -0,0 +1,405 @@
+// SPDX-FileCopyrightText: Intel Corporation
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "xhp-tests.hpp"
+
+template <typename... Rs> auto test_zip(Rs &&...rs) {
+  return dr::mhp::views::zip_base(std::forward<Rs>(rs)...);
```

Comment:
@suyashbakshi: `zip_base` is a zip that should be able to handle dot product. Can you try it?